modules/ecs-cluster/main.tf

```diff
@@ -0,0 +1,262 @@
+# ----------------------------------------------------------------------------------------------------------------------
+# Locals and Data Sources
+# ----------------------------------------------------------------------------------------------------------------------
+
+data "aws_elb_service_account" "current" {
+}
+
+data "aws_region" "current" {}
+
+data "aws_ami" "ecs" {
+
+  owners = ["amazon"]
+
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-ecs-hvm-2.0*-x86_64-ebs"]
+  }
+
+  most_recent = true
+}
+
+data "template_file" "user_data" {
+  template = file("${path.module}/templates/user_data.sh")
+
+  vars = {
+    aws_ecs_cluster_name    = aws_ecs_cluster.default.name
+    aws_ecs_default_region = data.aws_region.current.name
+  }
+}
+
+locals {
+  # This should be the same for all containers in other for clustering to work
+  parameters = {
+    URL                     = "https://platform.boomi.com"
+    BOOMI_ENVIRONMENT_NAME  = "PROD"
+    BOOMI_ENVIRONMENT_CLASS = "PRODUCTION"
+    BOOMI_CONTAINERNAME     = "ecs-mol"
+    BOOMI_ATOMNAME          = "ecs-atom"
+    INSTALLATION_DIRECTORY  = "/var/boomi"
+  }
+}
+
+# ----------------------------------------------------------------------------------------------------------------------
+# WS Resources
+# ----------------------------------------------------------------------------------------------------------------------
+resource "aws_ecr_repository" "this" {
+  name = join("-", [var.prefix, "ecr"])
+}
+
+resource "aws_ecs_cluster" "default" {
+  name = join("-", [var.prefix, "cls"])
+}
+
+resource "aws_kms_key" "default" {
+  description             = "For use on ${var.prefix}-kms"
+  deletion_window_in_days = 10
+}
+
+resource "aws_kms_alias" "default" {
+  name          = "alias/${var.prefix}-kms"
+  target_key_id = aws_kms_key.default.key_id
+}
+# ----------------------------------------------------------------------------------------------------------------------
+# S3 Logging Bucket
+# ----------------------------------------------------------------------------------------------------------------------
+resource "aws_s3_bucket" "bucket" {
```

Comment:
Logging s3 bucket should be provided as an input to the reusable module, maybe the caller wants to use another s3 bucket instead of creating a new logging bucket just for Boomi.